docs/js/features/odata/use-generated-client.md

```diff
@@ -0,0 +1,305 @@
+        )
+    )
+    .execute({
+        url: '<yourURL>'
+    });
+```
+
+The above translates to the following `expand` query parameters:
+
+```sql
+$select=FirstName,to_BusinessPartnerAddress/AddressID,to_BusinessPartnerAddress/CityCode&$expand=to_BusinessPartnerAddress
+```
+
+### Filter
+
+When operating on a collection of entities the API offers `filter( ... )` on the builders. It directly corresponds to the `$filter` parameter of the request. Filters are also build via the static property fields on entities.
+
+The following example:
+
+```ts
+/*
+Get all business partners that either:
+  - Have first name 'Alice' but not last name 'Bob'
+  - Or have first name 'Mallory'
+*/
+BusinessPartner.requestBuilder()
+    .getAll()
+    .filter(
+        or(
+            and(
+                BusinessPartner.FIRST_NAME.equals('Alice'),
+                BusinessPartner.LAST_NAME.notEquals('Bob')
+            ),
+            BusinessPartner.FIRST_NAME.equals('Mallory')
+        )
+    )
+    .execute({
+        url: '<yourURL>'
+    });
+```
+
+Will translate to this filter parameter:
+```sql
+$filter=(((FirstName eq 'Alice') and (LastName ne 'Bob')) or (FirstName eq 'Mallory'))
+```
+
+Take note of the order of `and` and `or`. As `or` is invoked on the result of `and` it will form the outer expression while `and` is an inner expression in the first branch of `or`.
+
+To achieve a different order with `and` as the top level statement one would nest the `or` within `and(...)`:
+
+```ts
+and(
+    or(
+        BusinessPartner.FIRST_NAME.equals('Alice'),
+        BusinessPartner.LAST_NAME.notEquals('Bob')
+    ),
+    BusinessPartner.FIRST_NAME.equals('Mallory')
+)
+```
+
+#### Available Filter Expressions
+
+<!-- TODO: Explain filters on complex types and navigational properties -->
+
+The [OData v4 standard](http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-protocol.html#sec_SystemQueryOptionfilter) allows for a wide range of filter expressions. A detailed list of what is available in the SDK can be obtained from [the documention](https://sap.github.io/cloud-sdk/docs/js/api-reference-js-ts/). The functionality can also be discovered through the fluent API.
+
+The below example leverages OData v4 exclusive features to build a more complex request:
+
+```ts
+/*
+Fetch all business partners where:
+- the last name is at least twice as long as the first name
+- AND the combined string of first and last name does not contain 'bob'
+*/
+BusinessPartner.requestBuilder()
+    .getAll()
+    .filter(
+        and(
+            filterFunction('multiply', 'int', filterFunction('length', 'int', BusinessPartner.FIRST_NAME), '2')
+                .lessThan(BusinessPartner.LAST_NAME.edmType.length)
+            ,substringOf(
+                filterFunction('concat', 'string', BusinessPartner.FIRST_NAME, BusinessPartner.LAST_NAME._fieldName),
+                'bob')
+                    .equals(true)
+        )
+    )
```

Comment:
This example is like a hammer on my head. I think this should be split into two smaller examples that we can take a closer look at in the accompanying text:

1. We have predefined fns, e. g. length, substring, substringOf (this can actually even be linked in the documentation - you will have to search for it otherwise you won't find it).
```ts
.filter(length(BusinessPartner.FIRST_NAME).lessThan(5))
```

2. You can construct other filter functions:
```
.filter(filterFunction('multiply', 'int', length(BusinessPartner.FIRST_NAME, '2').lessThan(5)))
```

Also the construction of additional filter functions is not explained in the text. We can also leave out the filter functions for now. and just add this as a comment in the file.

Reply:
@marikaner I can't use the length function, the types seem to be incompatible but I can't figure out the problem
![image](https://user-images.githubusercontent.com/66725443/85555372-2260c080-b626-11ea-9157-fd4312ee0246.png)